geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/CreateRegionCommandDUnitTest.java

```diff
@@ -15,290 +15,150 @@
-    assertEquals(Result.Status.OK, cmdResult.getStatus());
+  MemberVM locator;
+  MemberVM server;
 
-    // Make sure our region exists with compression enabled
-    vm.invoke(() -> {
-      Region region = getCache().getRegion("compressedRegion");
-      assertNotNull(region);
-      assertTrue(
-          SnappyCompressor.getDefaultInstance().equals(region.getAttributes().getCompressor()));
-    });
+  @Rule
+  public LocatorServerStartupRule lsRule = new LocatorServerStartupRule();
 
-    // cleanup
-    commandStringBuilder = new CommandStringBuilder(CliStrings.DESTROY_REGION);
-    commandStringBuilder.addOption(CliStrings.DESTROY_REGION__REGION, "compressedRegion");
-    cmdResult = executeCommand(commandStringBuilder.toString());
-    assertEquals(Result.Status.OK, cmdResult.getStatus());
-  }
+  @Rule
+  public GfshShellConnectionRule gfsh = new GfshShellConnectionRule();
 
-  /**
-   * Asserts that the "compressor" option for the "create region" command fails for an unrecognized
-   * compressorc.
-   */
-  @Test
-  public void testCreateRegionWithBadCompressor() {
-    setUpJmxManagerOnVm0ThenConnect(null);
+  @Rule
+  public TestName testName = new SerializableTestName();
 
-    VM vm = Host.getHost(0).getVM(1);
+  @Rule
+  public TemporaryFolder tmpDir = new TemporaryFolder();
 
-    // Create a cache in vm 1
-    vm.invoke(() -> assertNotNull(getCache()));
+  @Before
+  public void before() throws Exception {
+    locator = lsRule.startLocatorVM(0);
+    server = lsRule.startServerVM(1, locator.getPort());
 
-    // Create a region with an unrecognized compressor
-    CommandStringBuilder commandStringBuilder = new CommandStringBuilder(CliStrings.CREATE_REGION);
-    commandStringBuilder.addOption(CliStrings.CREATE_REGION__REGION, "compressedRegion");
-    commandStringBuilder.addOption(CliStrings.CREATE_REGION__REGIONSHORTCUT, "REPLICATE");
-    commandStringBuilder.addOption(CliStrings.CREATE_REGION__COMPRESSOR, "BAD_COMPRESSOR");
-    CommandResult cmdResult = executeCommand(commandStringBuilder.toString());
-    assertEquals(Result.Status.ERROR, cmdResult.getStatus());
+    gfsh.connectAndVerify(locator);
+  }
 
-    // Assert that our region was not created
-    vm.invoke(() -> {
-      Region region = getCache().getRegion("compressedRegion");
-      assertNull(region);
+  @Test
+  public void testCreateRegionWithGoodCompressor() throws Exception {
+    String regionName = testName.getMethodName();
+    gfsh.executeAndVerifyCommand("create region --name=" + regionName
+        + " --type=REPLICATE --compressor=" + RegionEntryContext.DEFAULT_COMPRESSION_PROVIDER);
+
+    server.invoke(() -> {
+      Cache cache = LocatorServerStartupRule.serverStarter.getCache();
+      Region region = cache.getRegion(regionName);
+      assertThat(region).isNotNull();
+      assertThat(SnappyCompressor.getDefaultInstance())
+          .isEqualTo(region.getAttributes().getCompressor());
```

Comment:
Seems to me the terms of this assertion are reversed. `region.getAttributes().getCompressor()` should be the actual and is expected to be `SnappyCompressor.getDefaultInstance()`.